rpc/backend/blocks.go

```diff
@@ -283,6 +284,51 @@ func (b *Backend) EthMsgsFromTendermintBlock(
 	return result
 }
 
+func (b *Backend) EthMsgsFromTendermintEndBlock(
+	blockRes *tmrpctypes.ResultBlockResults,
+) []*evmtypes.MsgEthereumTx {
+	var result []*evmtypes.MsgEthereumTx
+
+	for _, event := range blockRes.EndBlockEvents {
+		if event.Type != evmtypes.EventTypeEthereumTx {
+			continue
+		}
+		resp, err := b.parseTxDetailsFromEndBlocker(event)
```

Comment:
same for method name, parsing from event, not end blocker?

Reply:
It is parsing from end block and the only thing there is in the end block section of a block are events and technically we are parsing the txs that were applied in the end block which is reflected in the names.